float32 angleX
float32 angleY
float32 rotation
int32   count
string  color


